common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -739,6 +739,30 @@ bool mon_wlan_hal_dwpal::sta_beacon_11k_request(const SBeaconRequest11k &req, in
         cmd += " ssid=" + req_ssid;
     }
 
+    // use ap_ch_report in the request.
+    if (req.use_optional_ap_ch_report) {
+        std::stringstream ch_report_ss;
+        uint8_t current_channel = 0;
+        for (; current_channel < req.use_optional_ap_ch_report - 1; ++current_channel) {
```

Comment:
According to the yaml file, use_optional_ap_ch_report is a boolean. So I think this should be 
```suggestion
        for (; current_channel < sizeof(ap_ch_report) - 1; ++current_channel) {
            if (req.ap_ch_report[current_channel] == 0)
                break;
```

Or if you want to use it as a length field, rename it in the yaml.

Reply:
After discussion in zoom: let's just keep the structure as it is and use it as a length, and only change the comment in the yaml file so it says length instead of boolean.